ultravox/tools/ds_tool/ds_tool.py

```diff
@@ -175,6 +176,97 @@ def _map_sample(self, sample, exclude_fields):
         return sample
 
 
+@dataclasses.dataclass
+class AudioExtensionTask:
+    audio_column_name: str = simple_parsing.field(default="audio", alias="-a")
+    asr_column_name: str = simple_parsing.field(default="sentence", alias="-A")
+    translation_column_name: str = simple_parsing.field(
+        default="translation", alias="-T"
+    )
+    id_column_name: str = simple_parsing.field(default="id", alias="-i")
+    extend_type: str = simple_parsing.field(
+        default="repeat", alias="-e", choices=["repeat", "combine"]
+    )
+    multiplier: int = simple_parsing.field(default=2, alias="-m")
+
+    def map_split(
+        self,
+        ds_split: datasets.Dataset,
+        num_proc: int,
+        writer_batch_size: int,
+        exclude_fields: List[str],
+    ) -> datasets.Dataset:
+        print(
+            f'Extending audio using "{self.extend_type}" method with multiplier {self.multiplier}'
+        )
+
+        if self.extend_type == "repeat":
+            return ds_split.map(
+                function=self._map_sample_repeat,
+                num_proc=num_proc,
+                writer_batch_size=writer_batch_size,
+                remove_columns=ds_split.column_names,
+            )
+        elif self.extend_type == "combine":
+            return ds_split.map(
+                function=self._map_batch_combine,
+                batched=True,
+                batch_size=self.multiplier,
+                num_proc=num_proc,
+                writer_batch_size=writer_batch_size,
+                remove_columns=ds_split.column_names,
+            )
+        else:
+            raise ValueError(f"Unknown extend_type: {self.extend_type}")
+
+    def _map_sample_repeat(self, sample):
+        audio = sample[self.audio_column_name]
+        sentence = sample[self.asr_column_name]
+        translation = sample[self.translation_column_name]
+
+        if not isinstance(audio, dict) or "array" not in audio:
+            raise ValueError(f"Unsupported audio format: {type(audio)}")
+
+        audio_data = audio["array"]
+        repeated_audio = np.tile(audio_data, self.multiplier)
+        repeated_sentence = " ".join([sentence] * self.multiplier)
+        repeated_translation = " ".join([translation] * self.multiplier)
+
+        new_audio = {key: value for key, value in audio.items() if key != "path"}
```

Comment:
I think there are only 3 keys in an Audio, and the only one we care about copying is `sampling_rate`. Maybe just init the Audio object directly here?